paddle/fluid/operators/controlflow/while_op.cc

```diff
@@ -341,6 +365,52 @@ class WhileGradOpDescMaker : public framework::SingleGradOpDescMaker {
     // while operator could be renamed.
     while_grad->SetAttr("original_output_grad", output_grads_list);
 
+    /* The following codes are used in eager deletion mode */
+    std::unordered_set<std::string> bwd_skip_vars;
+    if (framework::GetEagerDeletionThreshold() >= 0) {
+      std::unordered_set<std::string> fwd_skip_vars;
+      for (auto *op_desc : grad_block->AllOps()) {
+        auto skippable = [&](const std::string &name) {
+          return !grad_block->HasVar(name) &&
+                 (fwd_block->HasVarRecursive(name) ||
+                  parent_block->HasVarRecursive(name));
+        };
+        for (auto &in_arg_name : op_desc->InputArgumentNames()) {
+          if (skippable(in_arg_name)) {
+            fwd_skip_vars.insert(in_arg_name);
+          }
+        }
+
+        for (auto &out_arg_name : op_desc->OutputArgumentNames()) {
+          if (skippable(out_arg_name)) {
+            fwd_skip_vars.insert(out_arg_name);
+          }
+        }
+      }
+
+      if (!fwd_skip_vars.empty()) {
+        // FIXME(zjl): ugly const_cast here, maybe we should find a better way
+        // to modify forward while_op
```

Comment:
I think the backward maker should be able to change the forward op desc by defined.

Sometimes, the forward op's behaviour is different when there is a backward op.